pkg/asset/machines/aws/instance_types.go

```diff
@@ -23,7 +24,50 @@ func PreferredInstanceType(ctx context.Context, meta *awsconfig.Metadata, types
 		return types[0], err
 	}
 
-	client := ec2.New(sess, aws.NewConfig().WithRegion(meta.Region))
+	found, err := getInstanceTypeZoneInfo(ctx, sess, meta.Region, types, zones)
+	if err != nil {
+		return types[0], err
+	}
+
+	reqZones := sets.NewString(zones...)
+	for _, t := range types {
+		if reqZones.Difference(sets.NewString(found[t]...)).Len() == 0 {
```

Comment:
I understand that this is existing code, but let's fix this.
```suggestion
		if sets.NewString(found[t]...).HasAll(zones...) {
```